packages/gov-app/libs/utils/pinProposalToIPFS.ts

```diff
@@ -0,0 +1,38 @@
+import { PINATA_JWT } from "@gov-app/libs/constants";
+
+interface ProposalData {
+	proposalTitle: string;
+	proposalDetails: string;
+}
+
+export const pinProposalToIPFS = async ({
+	proposalTitle,
+	proposalDetails,
+}: ProposalData) => {
+	const data = JSON.stringify({
+		pinataOptions: {
+			cidVersion: 1,
+		},
+		pinataContent: {
+			title: proposalTitle,
+			description: proposalDetails,
+		},
+		pinataMetadata: {
+			name: "proposal",
+		},
+	});
+
+	const response = await fetch(
+		"https://api.pinata.cloud/pinning/pinJSONToIPFS",
+		{
+			method: "POST",
+			headers: {
+				"Authorization": `Bearer ${PINATA_JWT}`,
+				"Content-Type": "application/json",
+			},
+			body: data,
+		}
+	);
+
+	return response.json();
```

Comment:
could do a little more error handling here, plus `response.json()` is an async operation

```suggestion
const data = await response.json();

if(!response.ok) {
  throw {code: response.status message: data?.message ?? response.statusText};
}

return data;
```